draft-meynell-panrg-scion-research-questions.md

```diff
@@ -201,11 +201,36 @@ This draft assumes the reader is familiar with some of the fundamental concepts
 * One AS belongs to many ISDs?
 
 
+## Beacon Forwarding Policy
+
+The idea behind "beaconing" is to discover all possible paths (loop free and with a fixed maximum length) between two CORE AS.
+Every AS forwards any received beacons to all neighbor CORE ASes unless this would cause a loop or exceed the fixed maximum length.
+Implemented naively, the number of paths (and beacons) grows exponentially with the network size.
+This is currently mitigated, primarily (and efficiently), by forwarding only the five "best PCBs" to neighbours.
+
+This "best PCB" practice has several implications:
+
+* It limits the number of available paths for endpoints such that only five truly diverse paths may be available to a given destination.
+  "truly diverse" means: Most ASes will receive five PCBs to a given destination from several of their neighbours. However, assuming
+  those ASes are use a similar PCB selection policy, it is not unlikely that each neighbor AS will forward a very
+  similar set of five PCB. In effect, there may be more than five paths available, but they are likely to diverge only on the first
+  one or two hops **(TBC: Is that true? There is no research confirming this hypothesis)**.
+  The problem that there may be only five truly diverse paths available is especially relevant for:
+  * Multipathing: endpoints may not be able to use more than five paths and no great selection of pats on offer.
+  * Unusual path policies, such as geofencing, may not be fulfillable by the limited number of paths on offer.
+* It causes a small number of paths to carry all traffic to a given destination, see {{link-load-balancing}}.
+* It opens the possibility of a DoS attack where a small number of "wormholes", see {{dos-with-wormholes}}.
+
+
 ## Segment Dissemination
 
-Control servers return a large number of path segments. This can cost considerable bandwidth / network egress while at the same time overloading clients with an unnecessarily large numbers of segments, mostly consisting of redundant information in terms of duplicate link and hops.
+Control servers return a large number of path segments when an endpoint requests paths to a destination.
+This can cost considerable bandwidth / network egress while at the same time
+overloading clients with an unnecessarily large numbers of segments,
+mostly consisting of redundant information in terms of duplicate links and hops.
 
-* This problem may be more problematic in ASes with many end hosts (e.g. IoT), or end hosts with little computing power or little spare bandwidth.
+* This problem may be more acute in ASes with many end hosts (e.g. IoT),
+or end hosts with little computing power or little spare bandwidth.
 * Getting a full path to a remote endhost may require three round-trips with the control server.
```

Comment:
Three round trips from the local control service to control services of other two ASes.

Reply:
I actually meant from the local endpoint to the local control server. I am less concerned about roundtrips between ASes, they have stonger machines and better connections.

Reply:
AFAIK the endpoint requests once the path to the destination, and receives up to three segments, all in one round trip.
The CS of the local AS may have to do several trips to core ASes in its local ISD, and then core ASes in the remote ISD.
But I think the general idea is understood, approving.